src/player.js

```diff
@@ -184,14 +191,25 @@ export default class Player extends FakeEventTarget {
     Utils.Object.mergeDeep(this._config, config);
     this._configureOrLoadPlugins(config.plugins);
     if (config.sources) {
-      this._maybeResetPlayer();
```

Comment:
`config.source` could be empty. `[]` consider as `true`

Reply:
sources is an object

Reply:
same potential problem in object